src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Collections;
+using System.Diagnostics;
+using System.Linq;
+
+namespace System.Text.Json.Serialization
+{
+    public static partial class JsonSerializer
+    {
+        private static void HandlePropertyName(
+            JsonSerializerOptions options,
+            ref Utf8JsonReader reader,
+            ref ReadStack state)
+        {
+            if (state.Current.Drain)
+                return;
+
+                Debug.Assert(state.Current.ReturnValue != default);
+                Debug.Assert(state.Current.JsonClassInfo != default);
+
+            if (state.Current.IsProcessingDictionary)
+            {
+                if (ReferenceEquals(state.Current.JsonClassInfo.DataExtensionProperty, state.Current.JsonPropertyInfo))
+                {
+                    ReadOnlySpan<byte> propertyName = reader.HasValueSequence ? reader.ValueSequence.ToArray() : reader.ValueSpan;
```

Comment:
What do we need to do to avoid the `.ToArray()` call here especially since its just an intermediary to the end result which is a UTF-16 transcoded string.

Reply:
In fact, why not just call `reader.GetString()` instead?

Reply:
The code needs both the UTF-8 (unescaped) since that is what it caches and eventually will get the UTF-16 for other reasons (to add to the extensiondata dictionary as a string)... So if it calls `reader.GetString()` first then it will have to encode it to UTF8 unescaped - is that any better? `reader.GetString()` also calls `ToArray`.

Also you'll see similar code in JsonDocument\Element.